tcl/filter.tcl

```diff
@@ -50,10 +50,13 @@ proc qc::filter_authenticate {event {error_handler qc::error_handler}} {
         # Check if this request is registered.
         if { [qc::registered $method $url_path] } {
             if {[qc::cookie_exists session_id] && [qc::session_valid [qc::session_id]]} {
-                qc::session_update [qc::session_id]
+                global session_id current_user_id
+                qc::session_update $session_id
+                set current_user_id [qc::session_user_id $session_id]
             } elseif {$method ni [list "GET" "HEAD"] && [qc::cookie_exists session_id] && ! [qc::session_valid [qc::session_id]]} {
```

Comment:
I think I'd prefer not to expose the global session_id here.

qc::session_id should be optomised to use the global session_id variable itself so I'd prefer to call ```qc::session_id``` multiple times or use a local variable to store it temporarily ```set session_id [qc::session_id]```